### DISCUSSION TITLE
Separate packages per WASM

### DISCUSSION THREAD

**Discussion by thewilkybarkid** [hpcc-systems/hpcc-js-wasm#76] [Ideas]
Thanks for the library, makes it a lot easier to use GraphViz.

Would it be possible to split the npm package into one per WASM? They can be fairly large, and I'm not using the other. This will shrink the size of the generated JS files (and their sourcemaps).

https://www.npmjs.com/package/node-graphviz exists, but I'd rather depend on the maintained library (and it seems like Node is supported now anyway).

Related, the test files are included in the package which adds a couple of MB.

**Comment by GordonSmith** [comment]
Thanks for your comments:
1. Is your size concern a runtime concern or a general package size?  FWIW removing the expat stubs would remove 55K from the minimized JS file (but it is a requirement for graphviz wasm - for the html support...).
2. I will remove the test files (as a quick win).

**Reply by thewilkybarkid** [reply]
Thanks. General package size: I'm only it using rather simplistically from Node (https://github.com/parcel-bundler/parcel/pull/6544).

**Reply by GordonSmith** [reply]
Hmmm this PR:  https://github.com/hpcc-systems/hpcc-js-wasm/pull/77 includes an additional dist/graphviz.js and dist/expat.js files which lets you manually import just the bit you need, but at the expense of a larger package (well I did remove the test files)...
